romana/cmd/tenant.go

```diff
@@ -108,28 +110,65 @@ func tenantCreate(cmd *cli.Command, args []string) error {
 		return err
 	}
 
-	tenants := []tenantData{}
+	tenants := []tenant.Tenant{}
 	for _, tnt := range args {
-		if !adaptor.TenantExists(tnt) {
-			// uncomment this once creating tenant for all
-			// platforms are ready. Till then tenants needs
-			// to be manually created for every platform and
-			// then created using romana command line tools.
-			// if adaptor.CreateTenant(tnt) != nil {
-			return err
-			// }
-		}
+		// uncomment this once creating tenant for all
+		// platforms are ready. Till then tenants needs
+		// to be manually created for every platform and
+		// then created using romana command line tools.
+		// if adaptor.CreateTenant(tnt) != nil {
+		// 	return err
+		// }
+		// adaptor.GetTenantUUID below wouldn't be needed once
+		// adaptor.CreateTenant is supported.
 		tntUUID, err := adaptor.GetTenantUUID(tnt)
 		if err != nil {
-			return err
+			switch err {
+			case util.ErrUnimplementedFeature:
+				// kubernetes doesnt support tenants yet so
+				// exception for kubernetes till CreateTenant,
+				// GetTenantUUID, etc are supported for it.
+				tntUUID = hex.EncodeToString(uuid.NewRandom())
+			default:
+				return err
+			}
 		}
-		data := tenant.Tenant{Name: tntUUID}
-		result := tenant.Tenant{}
+
+		data := tenant.Tenant{Name: tnt, UUID: tntUUID}
+		var result map[string]interface{}
 		err = client.Post(tenantURL+"/tenants", data, &result)
 		if err != nil {
 			return err
 		}
-		tenants = append(tenants, tenantData{result, tnt, []tenant.Segment{}})
+		_, tFound := result["UUID"]
```

Comment:
What are we looking for here? That the UUID is included in the result that's returned from the POST request? Why? Maybe some comments to explain this would be good.